services/web-server/src/servers/createApp.js

```diff
@@ -1,13 +1,20 @@
 const bodyParser = require('body-parser-graphql');
+const session = require('express-session');
+const MemoryStore = require('memorystore')(session);
 const compression = require('compression');
 const cors = require('cors');
 const express = require('express');
 const playground = require('graphql-playground-middleware-express').default;
 const passport = require('passport');
+const url = require('url');
 const credentials = require('./credentials');
 
 module.exports = async ({ cfg, strategies }) => {
   const app = express();
+  const store = new MemoryStore({
```

Comment:
Will using a memory store mean that users are logged out every time we restart the web-server process?  Could we create an `AzureStore` that uses an Azure table?  Would that end up hitting Azure for every GraphQL query (even though it's only needed for getCredentials)?

Reply:
> Will using a memory store mean that users are logged out every time we restart the web-server process?

Good question. I think it will and that's probably not the best design choice.

> Could we create an AzureStore that uses an Azure table?

Yes. I can look into that.

> Would that end up hitting Azure for every GraphQL query (even though it's only needed for getCredentials)?

Yes since we have `rolling: true` which might cause performance issues. Another way is to probably only regenerate the session when `getCredentials` is called.

Reply:
> Could we create an AzureStore that uses an Azure table?

@djmitche, Do we happen to use `AzureStore` somewhere else in the codebase? I'm trying to find some documentation on this so I can understand on how to implement this.

Reply:
I made up the name, so no.  We do use Azure tables, though.  References:
 * https://github.com/taskcluster/azure-entities
 * https://github.com/taskcluster/taskcluster/tree/master/libraries/azure#readme

example:
https://github.com/taskcluster/taskcluster/blob/d135176b954fe411faebccd0f13d051f7ead667c/services/secrets/src/data.js#L5-L15

https://github.com/taskcluster/taskcluster/blob/d135176b954fe411faebccd0f13d051f7ead667c/services/secrets/src/main.js#L37-L51

With that, the `Secret` object defined in `main.js` has methods `load`, `create`, `scan` etc.